pkg/shell/base_index.js

```diff
@@ -317,7 +317,16 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
             var forward_command = false;
             var data = event.data;
             var child = event.source;
-            if (!child || typeof data !== "string")
+            if (!child)
+                return;
+
+            /* If it's binary data just send it */
```

Comment:
Good point. And thinkng about this more we may have to go with the solution you proposed of having multiple WebSockets.